### PR TITLE
[portinglayer] bridge dynamic endpoint example

### PR DESCRIPTION
**Pull request by xshuqun**
* bridge dynamic endpoint example

### CODE REVIEW

File: component/common/application/matter/example/bridge_dm/matter_drivers.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "matter_events.h"
+#include <lwip/ip_addr.h>
+#include <platform/CHIPDeviceLayer.h>
+
+CHIP_ERROR matter_driver_bridge_light_init(void);
+void matter_driver_bridge_endpoint_assign(void);
+void matter_driver_bridge_setup_server(void);
+void matter_driver_bridge_recv_server(void);
```
Comment (step0035) [inline component/common/application/matter/example/bridge_dm/matter_drivers.h]: The src code for setup_server and recv_server is not found

**Review by step0035 — COMMENTED** [review]
I think a better idea would be to create a class for bridge, then implementing callbacks which will be called when a new device is bridged (add) or when a bridged device leaves (remove).

Also, I feel that using TCP might not be the best. Should we skip this TCP and work on ble-mesh-matter bridge?

File: component/common/application/matter/example/bridge_dm/matter_drivers.h (hunk 1 above)
Comment (step0035) [reply component/common/application/matter/example/bridge_dm/matter_drivers.h]: I think a better idea is to create a class for bridge, then implement callbacks which will be called when an endpoint joins or leaves.

**Comment by step0035** [comment]
Should we just skip using TCP as bridge? I don't think it is very suitable as TCP is not a network by itself. A ble-mesh-matter bridge will be more suitable.

File: component/soc/realtek/amebad/fwlib/usrcfg/rtl8721dhp_intfcfg.c
Comment (step0035) [inline component/soc/realtek/amebad/fwlib/usrcfg/rtl8721dhp_intfcfg.c:1]: this one i think don't need to change in the actual PR

**Comment by step0035** [comment]
In `matter_interactions.cpp`, under `PostDownlinkEvent`, might need to change `xQueueSendFromISR`. Add check first to see whether thread is running in ISR or normal user task.

File: component/.vscode/settings.json
Comment (SirRu24) [inline component/.vscode/settings.json:1]: Shouldn't we remove the .vscode folder?

File: project/realtek_amebaD_va0_example/inc/inc_hp/platform_opts.h
Comment (SirRu24) [inline project/realtek_amebaD_va0_example/inc/inc_hp/platform_opts.h:1]: Should we leave it as default (0) for CONFIG_EXAMPLE_MATTER_BRIDGE ?

File: component/common/application/matter/example/bridge_dm/README.md
Comment (SirRu24) [inline component/common/application/matter/example/bridge_dm/README.md:1]: Can give the PSRAM Usage part, the same as in [light_dm readme example](https://github.com/ambiot/ambd_matter/blob/main/component/common/application/matter/example/light_dm/README.md)

File: component/soc/realtek/amebad/fwlib/usrcfg/rtl8721dhp_intfcfg.c
Comment (SirRu24) [inline component/soc/realtek/amebad/fwlib/usrcfg/rtl8721dhp_intfcfg.c:1]: Let the psram attributes be FALSE, user can change based on the guide on readme

File: component/.vscode/settings.json
Comment (xshuqun) [reply component/.vscode/settings.json:1]: It has been removed

File: project/realtek_amebaD_va0_example/inc/inc_hp/platform_opts.h
Comment (xshuqun) [reply project/realtek_amebaD_va0_example/inc/inc_hp/platform_opts.h:1]: Yes, the config should be default 0, the platform_opts.h changes has been removed

File: component/common/application/matter/example/bridge_dm/README.md
Comment (xshuqun) [reply component/common/application/matter/example/bridge_dm/README.md:1]: the readme has been renew modified.

File: component/soc/realtek/amebad/fwlib/usrcfg/rtl8721dhp_intfcfg.c
Comment (xshuqun) [reply component/soc/realtek/amebad/fwlib/usrcfg/rtl8721dhp_intfcfg.c:1]: this configuration has already been removed.

**Review by SandyYeow — APPROVED** [review]

**Pull request merged**